brain/circuit.cpp

```diff
@@ -103,10 +103,20 @@ class Circuit::Impl
     {}
     virtual ~Impl() {}
 
+    virtual size_t getNumNeurons() const = 0;
+
     GIDSet getGIDs( const std::string& target ) const
     {
-        return brion::Target::parse(
-            _targetParsers, target.empty() ? _circuitTarget : target );
+        if( target.empty( ))
+        {
+            GIDSet gids;
+            for( size_t i = 0; i < getNumNeurons(); ++i )
+                gids.insert( i + 1 );
+            return gids;
+        }
+        else
```

Comment:
You have radically changed the semantics of what an empty target is, just look at the code and you'll realize it.
What's your intention here?